File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] arr, int target) {
        int n= arr.length;
        int l = 0;
        int r = n-1;
        while(l<=r){
            int mid = (l+r)/2;
            if(arr[mid]<target){
                l = mid+1;
            }
            else if(arr[mid]>target){
                r = mid-1;
            }
            else{
                return mid;
            }
        }

        return -1;
       
    }
}

File: Data Structures & Algorithms/next-greater-element-i/submission-0.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        l = []
        for i in range(len(nums1)):
            index = self.method1(nums1[i],nums2)
            maxi = -1
            for j in range(index,len(nums2)):
                if nums1[i]!=nums2[j]:
                    if nums2[j]>nums1[i]:
                        maxi = nums2[j]
                        l.append(nums2[j])
                        break
            if maxi==-1:
                l.append(-1)
        return l
    def method1(self,target: int,arr: List[int]) -> int:
        for i in range(len(arr)):
            if(arr[i]==target):
                return i
        return -1